doc/en/user/source/community/oauth2/index.rst

```diff
@@ -0,0 +1,151 @@
+the Google REST response against a valid ``access_token`` and converting it to a OAuth2 standard one.
+
+Other than this the most interesting part is the implementation of the base class ``GeoServerOAuth2SecurityConfiguration``.
+
+The latter contains the Google implementation of the ``OAuth2RestTemplate``.
+
+In the next section we will see how to install and configure the OAuth2 security filter on GeoServer authenticating against 
+Google OAuth2 Provider.
+
+Configure the Google authentication provider
+--------------------------------------------
+
+The first thing to do is to configure the OAuth2 Provider and obtain ``Client ID`` and ``Client Sectet`` keys.
+
+#. Obtain OAuth 2.0 credentials from the Google API Console.
+
+   Visit the `Google API Console <https://console.developers.google.com/>`_ to obtain OAuth 2.0 credentials such as a client ID and client secret 
+   that are known to both Google and your application. The set of values varies based on what type of application you are building. 
+   For example, a JavaScript application does not require a secret, but a web server application does.
+   
+    .. figure:: images/google1.jpg
+       :align: center
+
+#. Create a new ``OAuth client ID``
+
+   Click on the ``Client credentials`` context menu as shown in the figure below.
+   
+    .. figure:: images/google2.jpg
+       :align: center
+
+#. Configure a new ``Web application``
+
+   Add the ``Name`` and the ``Authorized redirect URIs`` like shown here below.
+   
+    .. note:: Always add two entries for each URI. One without the ending ``/`` and another one with it.
+   
+    .. figure:: images/google3.jpg
+       :align: center
+
+#. Take note of the ``Client ID`` and the ``Client Sectet``.
+
+   At the end of the procedure Google will show-up a small dialog box with the ``Client ID`` and the ``Client Sectet``.
+   Those info can be always accessed and updated from the `Google API Console <https://console.developers.google.com/>`_
+   
+    .. figure:: images/google4.jpg
+       :align: center
+
+#. Optionally customize the ``OAuth consent screen``.
+
+   At any time it is possible to update and customize the ``OAuth consent screen``. You can put here your logo, app name, ToS and so on.
+
+    .. figure:: images/google5.jpg
+       :align: center
+
+Configure the GeoServer OAuth2 filter
+-------------------------------------
+
+#. Start GeoServer and login to the web admin interface as the ``admin`` user.
+#. Click the ``Authentication`` link located under the ``Security`` section of
+   the navigation sidebar.
+
+    .. figure:: images/filter1.jpg
+       :align: center
+
+#. Scroll down to the ``Authentication Filters`` panel and click the ``Add new`` link.
+
+    .. figure:: images/filter2.jpg
+       :align: center
+
+#. Click the ``OAuth2`` link.
+
+    .. figure:: images/filter3.jpg
+       :align: center
+
+#. Fill in the fields of the settings form as follows:
```

Comment:
No, please don't make people copy those links from a screenshot (actually, could the oauth plugin just provide defaults so that people do not have to type/copy them at all?)... is there any page documenting them, just in case they might be changing in the future. Anyways, what I had to type was:
* https://accounts.google.com/o/oauth2/token
* https://accounts.google.com/o/oauth2/auth
* http://localhost:8080/geoserver (not sure if this is going to work, see google not accepting it as a redirect url before)
* https://www.googleapis.com/oauth2/v1/tokeninfo
* https://accounts.google.com/logout